Raven.Client.Lightweight/Document/RemoteBulkInsertOperation.cs

```diff
@@ -281,7 +280,7 @@ private void WriteQueueToServer(Stream stream, BulkInsertOptions options, Cancel
                         break;
                 }
 
-                FlushBatch(stream, batch);
+                return FlushBatch(stream, batch);
```

Comment:
You cannot exit here since you are still processing the queue (this would fail after more than 512 docs in `BulkInsert`)

Reply:
Good catch. Missed that